### PR TITLE
bug (core/highlights) : highlighting links in <pre> block

### PR DESCRIPTION
**Pull request by CodHeK**

### CODE REVIEW

**Comment by CodHeK** [comment]
@marcoscaceres this simple fix, works!

File: src/core/highlight.js [hunk 1]
```diff
@@ -58,7 +59,6 @@ export async function run(conf) {
         if (id !== msg.id) {
           return; // not for us!
         }
-        element.innerHTML = value;
```
Comment (marcoscaceres) [inline src/core/highlight.js]: Not sure how this works, as the `value` is supposed to be the highlighted HTML? What am I missing?

**Comment by CodHeK** [comment]
What I am thinking is, previously the whole of <pre> tag content is being replaced with the text in all the elements inside it which was the problem originally right ? So, I believe the right change would be to not do that ... 


On Sunday, February 17, 2019, 3:16 PM, Marcos Cáceres <notifications@github.com> wrote:


@marcoscaceres commented on this pull request.

In src/core/highlight.js:
> @@ -58,7 +59,6 @@ export async function run(conf) {
         if (id !== msg.id) {
           return; // not for us!
         }
-        element.innerHTML = value;

Not sure how this works, as the value is supposed to be the highlighted HTML? What am I missing?

**Comment by marcoscaceres** [comment]
> What I am thinking is, previously the whole of <pre> tag content is being replaced with the text in all the elements inside it which was the problem originally right ? So, I believe the right change would be to not do that ...

**Comment by marcoscaceres** [comment]
> What I am thinking is, previously the whole of `<pre>` tag content is being replaced with the text in all the elements inside it which was the problem originally right ? So, I believe the right change would be to not do that ...

The `value` is the highlighted code tho (i.e., it's the HTML the syntax highlighter produced). So, if we don't replace the `innerHTML`, then we don't get any highlights. So, you are correct... it does fix the issue, but it means we get no highlighted code :)

**Comment by CodHeK** [comment]
@marcoscaceres so, do we just proceed with using `class="nohighlight"` for such cases or re-write `respec-worker` to be able to handle such cases ?

**Comment by marcoscaceres** [comment]
We decided not to fix this. Thanks for investigating it @CodHeK.

**Pull request closed without merging**